linked_list/implementation.py

```diff
@@ -1,40 +1,235 @@
+from .node import Node
 from .interface import AbstractLinkedList
-
+from copy import deepcopy
 
 class LinkedList(AbstractLinkedList):
     """
     Implementation of an AbstractLinkedList inteface.
     """
 
     def __init__(self, elements=None):
-        pass
-
+        self.start = None
+        self.end = None
+        
+        if elements == None:
+            self.start = self.end
+        elif len(elements) == 1:
+            # Build a single node here
+            if isinstance(elements, list):
+                newnode = Node(elements[0])
+                self.start = newnode
+                self.end = newnode
+            else:
+                newnode = Node(elements)
+                self.start = newnode
+                self.end = newnode
```

Comment:
Nice job taking care of the case when someone doesnt assign a list as the parameter initially for a LinkedList